src/react/features/filters/hooks/useTextFilter.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect } from 'react'
+import { useDispatch, useStore } from '../../../store'
+import { useCompare } from '../../../utils/useCompare'
+
+import { setFocus } from '../../navigation/actions'
+import { addDiory, updateDiory } from '../../room/actions'
+
+const SEARCH_TOOL_ID = 'SEARCH_TOOL_ID'
+
+export const useTurnOnTextFilter = () => {
+  const [{ diograph }] = useStore((state) => state.room)
+  const [{ focus }] = useStore((state) => state.navigation)
+
+  const dispatch = useDispatch()
+  const turnOn = () => {
+    if (!diograph[SEARCH_TOOL_ID]) {
+      dispatch(addDiory({ id: SEARCH_TOOL_ID }))
+    }
+    if (SEARCH_TOOL_ID !== focus) {
+      dispatch(setFocus({ focus: SEARCH_TOOL_ID }))
+    }
+  }
+
+  return { turnOn }
+}
+
+export const useTextFilterValue = () => {
+  const [{ textFilter }] = useStore((state) => state.filters)
+  const [{ focus }] = useStore((state) => state.navigation)
+
+  return {
+    textFilterValue: SEARCH_TOOL_ID === focus ? textFilter : '',
+  }
+}
+
+const diographToLinks = (diograph, diory) => ({
+  ...diograph,
+  [diory.id]: {
+    id: diory.id,
+  },
+})
+
+const getFilteredLinks = (diograph, query = '') =>
+  Object.values(diograph)
+    .filter(({ id }) => id !== SEARCH_TOOL_ID)
+    .filter(({ text }) => !query || (text && text.toLowerCase().includes(query.toLowerCase())))
+    .reduce(diographToLinks, {})
```

Comment:
I liked the way these "non-exported" functions that are used only in one function would be defined inside of that function 😊 

Suggestion: Move `diographToLinks` and `getFilteredLinks` inside of `useTextFilter` block.

Reply:
I'll study this a bit...

Reply:
Now I remember :) they should not me inside because it's good that functions are independent and get data only as arguments. Inside other funcs they may get data directly from parent: source of errors and makes code more complicated.

Reply:
Oh, if that is the case, they really shouldn't be inside! 

Good job, thanks for making this clear. 💪👍